python_scripts/functions.py

```diff
@@ -407,28 +410,38 @@ def generate_iam_config(
     iam_policy_path: str
         Optional path to output the iam policy json generated from the iam_config just generated
     """
+    if os.path.exists(iam_config_output) and overwrite_config is not True:
```

Comment:
Would you not want to validate the config first?

Reply:
This function takes in a dictionary of the contents of the config file that has already been validated. Might be better to take in the actual config and validate within though, good point.

Reply:
I've left it as needing to take a ready validated config for now, but this will be pretty easy to change in future if we decide it is better to validate within this function.